app/src/vanilla/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -0,0 +1,619 @@
+    pauseDownload();
+    resumeDownload();
+    cancelDownload();
+    loadDownloadApp();
+  }
+
+  private void handleFirstLoad() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .doOnNext(__ -> view.showLoading())
+        .flatMap(__ -> loadApp())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, throwable -> crashReport.log(throwable));
+  }
+
+  private void handleReviewAutoScroll() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.scrollReviewsResponse())
+        .flatMap(reviews -> scheduleAnimations(reviews))
+        .subscribe(__ -> {
+        }, throwable -> crashReport.log(throwable));
+  }
+
+  private void handleClickOnScreenshot() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.getScreenshotClickEvent())
+        .filter(event -> !event.isVideo())
+        .doOnNext(imageClick -> {
+          appViewAnalytics.sendOpenScreenshotEvent();
+          appViewNavigator.navigateToScreenshots(imageClick.getImagesUris(),
+              imageClick.getImagesIndex());
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnVideo() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.getScreenshotClickEvent())
+        .filter(event -> event.isVideo())
+        .doOnNext(videoClick -> {
+          appViewAnalytics.sendOpenVideoEvent();
+          appViewNavigator.navigateToUri(videoClick.getUri());
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnDescriptionReadMore() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.clickedReadMore())
+        .doOnNext(readMoreClickEvent -> {
+          appViewAnalytics.sendReadMoreEvent();
+          appViewNavigator.navigateToDescriptionReadMore(readMoreClickEvent.getStoreName(),
+              readMoreClickEvent.getDescription(), readMoreClickEvent.getStoreTheme());
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnDeveloperWebsite() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.clickDeveloperWebsite())
+        .flatMapSingle(__ -> appViewManager.getDetailedAppViewModel(appId, packageName))
+        .filter(app -> !TextUtils.isEmpty(app.getDetailedApp()
+            .getDeveloper()
+            .getWebsite()))
```

Comment:
Does it make sense to filter out empty strings here? why is this section shown/clickable if it is empty?

Reply:
Also need to be careful using TextUtils - which is android specific.